packages/py3_pre_commit.rb

```diff
@@ -0,0 +1,15 @@
+require 'buildsystems/pip'
+
+class Py3_pre_commit < Pip
+  description 'A framework for managing and maintaining multi-language pre-commit hooks.'
+  homepage 'https://pre-commit.com/'
+  @_ver = '3.6.2'
+  version "#{@_ver}-py3.12"
+  license 'MIT'
+  compatibility 'all'
+  source_url 'SKIP'
+
+  depends_on 'python3'
+
+  no_compile_needed
```

Comment:
Add  postinstall  to execute  pre-commit?
```
def self.postinstall
  system 'pre-commit install'
end
```
Likewise, it might make sense to add `def self.remove` to clean up the pre-commit hooks installed locally unless they can be included with this package?

Reply:
We can't ship pre-commit hooks, and you have to run pre-commit install in the directory of a git repo.

Reply:
Why not?  What is the harm in doing so?  Nobody is going to know they need to run `pre-commit install`.  Might as well do it for them because, otherwise, this package is useless.  The linter package was created to make it EASIER for those who want to lint stuff.  This makes it harder.  One of my big pet peeves about changing things.

Reply:
running `pre-commit install` like this just doesn't work. thats not how the package works.

you run `pre-commit install` in a specific git directory when you want to set up the pre-commit hooks for that project.

And even still-- nobody is going to be installing this package without knowing to run `pre-commit install` in the relevant directory. If they're installing the `py3-pre-commit` package, they know what it does. This is a developer tool-- trying to guess what the user is aiming to do and doing it for them is not a good idea.

Reply:
The `CREW_LOCAL_REPO_ROOT` constant takes the guesswork out of the equation.

Reply:
Nobody is going to understand that they need to run `pre-commit install` unless you explain it to them.  Again, try and make things easier, not harder.

Reply:
But then we're assuming that users are installing this for the purpose of contributing to crew-- if they arent, we've installed those hooks (and the linters they use!) for no reason.

Reply:
The fundamental point of what I'm saying is that we shouldn't assume what a user wants to do and then do it for them.

Reply:
> Nobody is going to understand that they need to run `pre-commit install` unless you explain it to them. Again, try and make things easier, not harder.

Nobody is going to install the `py3-pre-commit` package without knowing what it does, and if they don't know and want to find out, they can use the description, or the homepage, or google.

We don't do this for any other package-- and neither do any other package managers.

Why should we do it for this one?

Reply:
We're not doing anything for them other than aiding the process.  Again, this package is USELESS without running `pre-commit install`.  And we don't have to include it in buildessential.  We've established that, at least.

Reply:
> We're not doing anything for them other than aiding the process.

No, we're assuming that they want to have linters installed in `CREW_LOCAL_REPO_ROOT` and doing that automatically for them. That's not aiding the process, that's at best doing something that they didn't ask for and don't understand, and at worst doing something completely unnecessary that runs counter to what they were trying to do.

> Again, this package is USELESS without running `pre-commit install`.

What exactly do you mean useless? This package provides a tool-- it is not useless just because we don't automatically use it.